cmd/xl-v1-multipart.go

```diff
@@ -1077,13 +1077,51 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 	return objInfo, nil
 }
 
+// Wrapper which removes all the uploaded parts.
+func (xl xlObjects) cleanupUploadedParts(bucket, object, uploadID string, storageDisks []StorageAPI) error {
+	var errs = make([]error, len(storageDisks))
+	var wg = &sync.WaitGroup{}
+
+	// Construct uploadIDPath.
+	uploadIDPath := path.Join(bucket, object, uploadID)
+
+	// Cleanup uploadID for all disks.
+	for index, disk := range storageDisks {
+		if disk == nil {
+			errs[index] = traceError(errDiskNotFound)
+			continue
+		}
+		wg.Add(1)
+		// Cleanup each uploadID in a routine.
+		go func(index int, disk StorageAPI) {
+			defer wg.Done()
+			err := cleanupDir(disk, minioMetaMultipartBucket, uploadIDPath)
+			if err != nil && errorCause(err) != errVolumeNotFound {
+				errs[index] = err
+			}
+		}(index, disk)
+	}
+
+	// Wait for all the cleanups to finish.
+	wg.Wait()
+
+	// Do we have write quorum?
+	if !isDiskQuorum(errs, xl.writeQuorum) {
```

Comment:
``isDiskQuorum`` looks suspect. It counts "non-ignored" errors and checks if the total is greater than the supplied quorum value. This is the opposite of what we want. We wish to know if there were quorum or more success (i.e, ``err == nil``) to deem the operation a success. This is not related to this patch, but this change relies on it.